lib/png/png.go

```diff
@@ -0,0 +1,119 @@
+package png
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	_ "embed"
+
+	"github.com/playwright-community/playwright-go"
+	"oss.terrastruct.com/d2/lib/xmain"
+)
+
+type Playwright struct {
+	PW      *playwright.Playwright
+	Browser playwright.Browser
+	Page    playwright.Page
+}
+
+func (pw *Playwright) RestartBrowser() (newPW Playwright, err error) {
+	if err = pw.Browser.Close(); err != nil {
+		return Playwright{}, err
+	}
+	return startPlaywright(pw.PW)
+}
+
+func (pw *Playwright) Cleanup() error {
+	if err := pw.Browser.Close(); err != nil {
+		return err
+	}
+	if err := pw.PW.Stop(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func startPlaywright(pw *playwright.Playwright) (Playwright, error) {
+	browser, err := pw.Chromium.Launch()
+	if err != nil {
+		return Playwright{}, err
+	}
+	context, err := browser.NewContext()
+	if err != nil {
+		return Playwright{}, err
+	}
+	page, err := context.NewPage()
+	if err != nil {
+		return Playwright{}, err
+	}
+	return Playwright{
+		PW:      pw,
+		Browser: browser,
+		Page:    page,
+	}, nil
+}
+
+func InitPlaywright() (Playwright, error) {
```

Comment:
why do you need to reimplement this? Why can't `playwright.Install` be used?

Reply:
their source code always prints out a "installing..." even when it's not.

Reply:
this needs a comment then. and can you link me to where in the source this is?

Reply:
@nhooyr It was here https://github.com/playwright-community/playwright-go/blob/8e8f670b5fa7ba5365ae4bfc123fea4aac359763/run.go#L92 but I missed that you can disable the Verbose flag to not show that line of text. Implementing that instead